nixd/lib/Server/Controller.cpp

```diff
@@ -204,9 +205,15 @@ void Controller::fetchConfig() {
         lspserver::ConfigurationParams{
             std::vector<lspserver::ConfigurationItem>{
                 lspserver::ConfigurationItem{.section = "nixd"}}},
-        [this](llvm::Expected<configuration::TopLevel> Response) {
+        [this](llvm::Expected<OptionalValue> Response) {
           if (Response) {
-            updateConfig(std::move(Response.get()));
+            llvm::Expected<configuration::TopLevel> ResponseConfig =
```

Comment:
Will this actually returns error in case of emacs' reply?

I'm just wondering if we received `[{}]` here why it uses JSONConfig as defaults.

Reply:
I have tested this with Emacs and it works without errors. Previously, if the language server received `[{}]`, it would use the defaults of the default constructor, not giving any errors. My changes just make it use a given default (the JSONConfig) instead (line 213).